### PR TITLE
8285687 : Remove jtreg tag manual=yesno for java/awt/print/PrinterJob/PageRangesDlgTest.java

### PR DESCRIPTION
**Pull request by lawrence-andrew**
1) Fixed Parser error by removing  yesno from @run main/manual=yesno 
2) Used PassFaileJFrame to show the test instruction to the user instead of printing the test instruction on the console or jtreg log file. 
3) If printer is not configured then mark the test as passed.

@shurymury 
@aivanov-jdk

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8285687](https://bugs.openjdk.java.net/browse/JDK-8285687): Remove jtreg tag manual=yesno for java/awt/print/PrinterJob/PageRangesDlgTest.java

### Reviewers
 * [Phil Race](https://openjdk.java.net/census#prr) (@prrace - **Reviewer**)
 * [Alexey Ivanov](https://openjdk.java.net/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8412/head:pull/8412` \
`$ git checkout pull/8412`

Update a local copy of the PR: \
`$ git checkout pull/8412` \
`$ git pull https://git.openjdk.java.net/jdk pull/8412/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8412`

View PR using the GUI difftool: \
`$ git pr show -t 8412`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8412.diff">https://git.openjdk.java.net/jdk/pull/8412.diff</a>

</details>

### CODE REVIEW

**Comment by mrserb** [comment]
The question about this and other tests is, should the test pass if the printer is not installed? I guess the printer is one of the pre-requirements for such tests and it should be configured properly.

**Comment by GoeLin** [comment]
Hi,  if this needs a printer, you might want to add @key printer? Best regards, Goetz

**Review by prrace — APPROVED** [review]

**Comment by mrserb** [comment]
Since the test now uses the printer keyword I doubt that it should pass silently if the printer is not installed.

**Comment by mrserb** [comment]
> @mrserb, run the test on Ubuntu 20.04 where printer is not configured and test passes as shown in the jtreg jtr file

That's my point, why does the manual test for the printer silently pass if the printer is not installed.

**Comment by prrace** [comment]
One thing we could start to do is use jtreg.SkippedException - which you define yourself.
It is a "special" in that jtreg recognises it as meaning a test was skipped due to some reason
that could not/was not determined before running. 
The test passes but with a special message so you know it didn't really run the test.
https://openjdk.java.net/jtreg/faq.html#what-if-a-test-does-not-apply-in-a-given-situation

For manual tests it might be good to change PassFailJFrame to be able to put up a  special message to the user in such a case and  disable "fail" ..

**Comment by aivanov-jdk** [comment]
Can't we use existing `jtreg.SkippedException` in `test/lib`?

Modify PageRangesDlgTest:

```diff
diff --git a/test/jdk/java/awt/print/PrinterJob/PageRangesDlgTest.java b/test/jdk/java/awt/print/Pri
nterJob/PageRangesDlgTest.java
index 77be99107d1..0d8ce30057a 100644
--- a/test/jdk/java/awt/print/PrinterJob/PageRangesDlgTest.java
+++ b/test/jdk/java/awt/print/PrinterJob/PageRangesDlgTest.java
@@ -37,9 +37,9 @@ import jtreg.SkippedException;
  * @bug 8061267
  * @key printer
  * @library /java/awt/regtesthelpers
- * @library /java/awt/jtreg
+ * @library /test/lib
  * @build PassFailJFrame
- * @build SkippedException
+ * @build jtreg.SkippedException
  * @summary The specified page range should be displayed in the dialog
  * @run main/manual PageRangesDlgTest
  */
```

The test can't pass when the native Print dialog is displayed: `aset` should be passed to it:
```diff
@@ -58,7 +58,7 @@ public class PageRangesDlgTest implements Printable {
         job = PrinterJob.getPrinterJob();
         job.setPrintable(new PageRangesDlgTest());
         aset.add(DialogTypeSelection.NATIVE);
-        if (job.printDialog()) {
+        if (job.printDialog(aset)) {
             job.print();
         }
     }
```

File: test/jdk/java/awt/print/PrinterJob/PageRangesDlgTest.java [hunk 1]
```diff
@@ -69,8 +63,32 @@ public static void main(String args[]) throws Exception {
         }
     }
 
+    public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length > 0) {
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/print/PrinterJob/PageRangesDlgTest.java:67]: The inverted condition produces cleaner code with less indentation:

```java
        if (PrinterJob.lookupPrintServices().length == 0) {
            throw new SkippedException("Printer not configured or available."
                                       + " Test cannot continue.");
        }

        // Rest of the test code
```

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**